backend/src/main/java/ch/puzzle/pcts/mapper/MemberOverviewMapper.java

```diff
@@ -0,0 +1,120 @@
+package ch.puzzle.pcts.mapper;
+
+import ch.puzzle.pcts.dto.memberoverview.*;
+import ch.puzzle.pcts.dto.memberoverview.certificate.MemberOverviewCertificateDto;
+import ch.puzzle.pcts.dto.memberoverview.certificate.MemberOverviewCertificateTypeDto;
+import ch.puzzle.pcts.dto.memberoverview.degree.MemberOverviewDegreeDto;
+import ch.puzzle.pcts.dto.memberoverview.degree.MemberOverviewDegreeTypeDto;
+import ch.puzzle.pcts.dto.memberoverview.experience.MemberOverviewExperienceDto;
+import ch.puzzle.pcts.dto.memberoverview.experience.MemberOverviewExperienceTypeDto;
+import ch.puzzle.pcts.dto.memberoverview.leadershipexperience.MemberOverviewLeadershipExperienceDto;
+import ch.puzzle.pcts.dto.memberoverview.leadershipexperience.MemberOverviewLeadershipExperienceTypeDto;
+import ch.puzzle.pcts.model.certificatetype.CertificateKind;
+import ch.puzzle.pcts.model.memberoverview.MemberOverview;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MemberOverviewMapper {
+
+    public MemberOverviewDto toDto(List<MemberOverview> memberOverviews) {
+
+        if (memberOverviews == null || memberOverviews.isEmpty()) {
+            return null;
+        }
+
+        MemberOverviewMemberDto memberDto = getMemberDto(memberOverviews);
+
+        List<MemberOverviewDegreeDto> degrees = memberOverviews
+                .stream()
+                .filter(e -> e.getDegreeId() != null)
```

Comment:
that looks not very maintainable 
pls check if you can extract logic into helpers or user other ways to increase readability